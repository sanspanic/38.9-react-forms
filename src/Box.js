import React, { useState } from "react";

const Box = ({ width, height, color }) => {
  return (
    <div
      style={{
        backgroundColor: color,
        height: `${height}px`,
        width: `${width}px`,
      }}
    ></div>
  );
};

export default Box;
