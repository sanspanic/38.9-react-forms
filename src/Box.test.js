import React from "react";
import { render } from "@testing-library/react";
import Box from "./Box";

test("ok: renders without crashing", function () {
  render(<Box />);
});

test("ok: matches snapshot", function () {
  const { asFragment } = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
});

//adds new box
function addBox(boxList, height = "20", width = "20", color = "red") {
  const heightInput = boxList.getByLabelText("Height");
  const widthInput = boxList.getByLabelText("Width");
  const backgroundInput = boxList.getByLabelText("Color");
  fireEvent.change(backgroundInput, { target: { value: color } });
  fireEvent.change(widthInput, { target: { value: width } });
  fireEvent.change(heightInput, { target: { value: height } });
  const button = boxList.getByText("Add a new box!");
  fireEvent.click(button);
}
