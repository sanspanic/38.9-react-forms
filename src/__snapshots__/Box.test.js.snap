// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ok: matches snapshot 1`] = `
<DocumentFragment>
  <div />
  <button>
    X
  </button>
</DocumentFragment>
`;
